pypeit/pypeit.py

```diff
@@ -431,8 +441,11 @@ def reduce_all(self):
                     science_basename[j] = self.basename
 
                     # TODO come up with sensible naming convention for save_exposure for combined files
-                    self.save_exposure(frames[0], sci_spec2d, sci_sobjs,
-                                       self.basename, history)
+                    if len(sci_spec2d.detectors) > 0:
+                        self.save_exposure(frames[0], sci_spec2d, sci_sobjs, self.basename, history)
+                    else:
+                        msgs.warn('No spec2d and spec1d saved to file because the '
+                                  'calibration/reduction was not successful for all the detectors')
```

Comment:
Not successful for all or for any?

Reply:
It's "not successful for all". If for some reason, some calibration step failed for all the detectors, `self.save_exposure` would crash.